ProcessMaker/Models/Setting.php

```diff
@@ -370,4 +372,104 @@ public static function getFavicon()
 
         return $url . '?id=' . bin2hex(random_bytes(16));
     }
+
+    /**
+     * Get the groups related to the specific group_id
+     *
+     * @param int menuId
+     *
+     * @return array
+     */
+    public static function groupsByMenu($menuId)
+    {
+        $query = Setting::query()
+            ->select('group')
+            ->groupBy('group')
+            ->where('group_id', $menuId)
+            ->notHidden()
+            ->pluck('group');
+        $response = $query->toArray();
+        $result = [];
+        foreach ($response as &$value) {
+            // Technical debts: we need to add int key to identify a group, currently this is a label
+            $result[] = [
+                'id' => $value,
+                'name' => $value,
```

Comment:
Consider to add a `group_label` field for the UI text.

```suggestion
                'name' => $value,
                'group_label' => $value,
```